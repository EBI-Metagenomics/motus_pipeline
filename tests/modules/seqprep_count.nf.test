nextflow_process {

    name "Test Process SEQPREP_REPORT"
    script "modules/seqprep.nf"
    process "SEQPREP_REPORT"

    test("Overlap reads count") {

        when {
            params {
                reads = "${baseDir}/tests/modules/fixtures/bwa/"
            }
            process {
                """
                input[0] = channel.fromPath("${params.reads}/mgtest_1.fastq.gz")
                input[1] = channel.fromPath("${params.reads}/mgtest_2.fastq.gz")
                input[2] = channel.fromPath("${params.reads}/mgtest_merged.fastq.gz")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}