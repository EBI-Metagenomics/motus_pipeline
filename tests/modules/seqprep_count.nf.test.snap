{
    "Overlap reads count": {
        "content": [
            {
                "overlapped_report": [
                    "output_report.txt:md5,2373ae841302743817122e4bbc3da010"
                ]
            }
        ],
        "timestamp": "2023-03-30T12:59:25+0000"
    }
}